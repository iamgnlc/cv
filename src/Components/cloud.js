const cloud = [
  { text: "PHP", value: 20 },
  { text: "Laravel", value: 15 },
  { text: "Lumen", value: 10 },
  { text: "Ruby On Rails", value: 10 },
  { text: "MySQL", value: 6 },
  { text: "MS SQL Server", value: 5 },
  { text: "Postgres", value: 6 },
  { text: "mongoDB", value: 3 },
  { text: "React", value: 9 },
  { text: "Node.js", value: 6 },
  { text: "Sails.js", value: 6 },
  { text: "HTML", value: 9 },
  { text: "XHTML", value: 5 },
  { text: "CSS", value: 8 },
  { text: "LESS", value: 8 },
  { text: "SASS", value: 8 },
  { text: "AJAX ", value: 3 },
  { text: "Javascript ", value: 8 },
  { text: "jQuery ", value: 4 },
  { text: "Pug/Jade ", value: 4 },
  { text: "HAML ", value: 4 },
  { text: "Foundation ", value: 4 },
  { text: "Bootstrap ", value: 5 },
  { text: "Gulp", value: 2 },
  { text: "Webpack", value: 2 },
  { text: "Yarn", value: 2 },
  { text: "Git", value: 10 },
  { text: "NPM", value: 2 },
  { text: "AWS", value: 8 },
  { text: "LAMP", value: 8 },
  { text: "Drupal", value: 9 },
  { text: "Wordpress", value: 9 },
  { text: "MS-DOS", value: 6 },
  { text: "Windows", value: 6 },
  { text: "Linux", value: 6 },
  { text: "Mac OS", value: 6 },
  { text: "SSH", value: 7 },
  { text: "CRON", value: 7 },
  { text: "Bash", value: 7 },
  { text: "Nginx", value: 7 },
  { text: "Apache", value: 7 },
  { text: "Varnish", value: 5 },
  { text: "Memcache", value: 5 },
  { text: "Redis", value: 5 },
  { text: "VM Ware", value: 5 },
  { text: "VirtualBox", value: 5 },
  { text: "Vagrant", value: 5 },
  { text: "PhotoShop", value: 3 },
  { text: "Illustrator", value: 3 },
]

export default cloud