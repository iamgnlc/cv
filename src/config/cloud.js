const cloud = [
  { text: "PHP", value: 20 },
  { text: "Laravel", value: 15 },
  { text: "Lumen", value: 10 },
  { text: "Ruby On Rails", value: 10 },
  { text: "MySQL", value: 7 },
  { text: "MS SQL Server", value: 5 },
  { text: "Postgres", value: 7 },
  { text: "mongoDB", value: 6 },
  { text: "React", value: 12 },
  { text: "Redux", value: 7 },
  { text: "Algolia", value: 5 },
  { text: "Node.js", value: 7 },
  { text: "Sails.js", value: 7 },
  { text: "HTML", value: 9 },
  { text: "XML", value: 5 },
  { text: "Xpath", value: 5 },
  { text: "CSS", value: 8 },
  { text: "LESS", value: 8 },
  { text: "SASS", value: 8 },
  { text: "AJAX ", value: 3 },
  { text: "API ", value: 4 },
  { text: "REST ", value: 4 },
  { text: "SOAP ", value: 4 },
  { text: "Javascript ", value: 8 },
  { text: "jQuery ", value: 4 },
  { text: "Pug/Jade ", value: 4 },
  { text: "HAML ", value: 4 },
  { text: "Foundation ", value: 4 },
  { text: "Bootstrap ", value: 5 },
  { text: "Git", value: 10 },
  { text: "SFTP", value: 2 },
  { text: "FTP", value: 2 },
  { text: "Bower", value: 2.5 },
  { text: "Gulp", value: 2.5 },
  { text: "Webpack", value: 2.5 },
  { text: "Yarn", value: 2.5 },
  { text: "NPM", value: 2.5 },
  { text: "AWS", value: 8 },
  { text: "Heroku", value: 6 },
  { text: "LAMP", value: 8 },
  { text: "Drupal", value: 9 },
  { text: "Wordpress", value: 9 },
  { text: "MS-DOS", value: 6 },
  { text: "Windows", value: 6 },
  { text: "Linux", value: 6 },
  { text: "Mac OS", value: 6 },
  { text: "Raspberry PI", value: 5 },
  { text: "SSH", value: 6 },
  { text: "CRON", value: 6 },
  { text: "Bash", value: 6 },
  { text: "Nginx", value: 6 },
  { text: "Apache", value: 6 },
  { text: "IIS", value: 5 },
  { text: "Varnish", value: 5 },
  { text: "Memcache", value: 5 },
  { text: "Redis", value: 5 },
  { text: "VM Ware", value: 5 },
  { text: "VirtualBox", value: 5 },
  { text: "Vagrant", value: 5 },
  { text: "PhotoShop", value: 3 },
  { text: "Illustrator", value: 3 },
  { text: "Ableton Live", value: 3 },
]

export default cloud
