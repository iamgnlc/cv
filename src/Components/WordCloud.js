import React, { PureComponent } from "react";
import TagCloud from "react-tag-cloud";

import cloud from "../config/cloud.js";

const cloudStyle = {
  fontFamily: "Inconsolata",
  width: "100%",
  height: "calc(100vh - 100px)"
};

class WordCloud extends PureComponent {
  shuffle = () => {
    this.forceUpdate();
  };

  autoShuffle = () => {
    setInterval(() => {
      this.shuffle();
    }, process.env.REACT_APP_AUTO_SHUFFLE_TIMEOUT);
  };

  componentDidMount() {
    // this.autoShuffle()
  }

  renderWord = (text, style) => {
    return (
      <span key={text} className="tag" style={style}>
        {text}
      </span>
    );
  };

  renderDesktop = () => {
    return (
      <TagCloud
        onClick={this.shuffle}
        className="d-none d-sm-block"
        style={cloudStyle}
      >
        {cloud.map((word, index) => {
          let tagStyle = {
            fontSize: word.value * 6,
            opacity: word.value / 10,
            padding: word.value * 1.5,
            zIndex: -index
          };
          return this.renderWord(word.text, tagStyle);
        })}
      </TagCloud>
    );
  };

  renderMobile = () => {
    return (
      <TagCloud
        onClick={this.shuffle}
        className="d-block d-sm-none"
        style={cloudStyle}
      >
        {cloud.map((word, index) => {
          let tagStyle = {
            fontSize: word.value * 3,
            opacity: word.value / 10,
            padding: word.value * 0.75,
            zIndex: -index
          };
          return this.renderWord(word.text, tagStyle);
        })}
      </TagCloud>
    );
  };

  render() {
    return (
      <React.Fragment>
        {this.renderDesktop()}
        {this.renderMobile()}
      </React.Fragment>
    );
  }
}

export default WordCloud;
