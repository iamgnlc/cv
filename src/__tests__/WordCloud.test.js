import React from "react"
import renderer from "react-test-renderer"

import "../config/cloud.js"

import WordCloud from "../Components/WordCloud"

describe("wordcloud component", () => {
  it("should render without crashing", () => {
    const tree = renderer.create(<WordCloud />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
