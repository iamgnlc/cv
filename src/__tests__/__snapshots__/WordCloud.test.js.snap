// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wordcloud component should render without crashing 1`] = `
Array [
  <div
    className="d-none d-sm-block"
    onClick={[Function]}
    style={
      Object {
        "height": "calc(100vh - 100px)",
        "width": "100%",
      }
    }
  />,
  <div
    className="d-block d-sm-none"
    onClick={[Function]}
    style={
      Object {
        "height": "calc(100vh - 100px)",
        "width": "100%",
      }
    }
  />,
]
`;
